src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java

```diff
@@ -308,14 +303,20 @@ void handleInvocationRequest(final int invId, final InputStream input) throws IO
                 int transactionId = unmarshaller.readUnsignedShort();
                 identity = identityId == 0 ? connection.getLocalIdentity() : connection.getLocalIdentity(identityId);
                 transaction = null; // todo: look up transaction by transactionId
+                locator = unmarshaller.readObject(EJBLocator.class);
```

Comment:
Moving these methods changes the protocol (see protocol.txt).  I think the app/mod name should come first, as described in that doc. Did you encounter a reason to change it?